lib/DxcSupport/HLSLOptions.cpp

```diff
@@ -69,6 +69,116 @@ const OptTable * hlsl::options::getHlslOptTable() {
   return g_HlslOptTable;
 }
 
+static std::vector<ArgPair> ComputeArgPairsFromArgList(const llvm::opt::InputArgList &argList) {
```

Comment:
FWIW, it's a shame there's not a single set of unit tests I can look at that directly exercise all the various cases and edge cases that `ComputeArgPairsFromArgList` handles.  Presumably there's good coverage for all 5 or so types of arguments this supports through the rest of the test suite?

Reply:
Actually we do in the new test case. -Zi and -Od are ::FlagClass, -T and -E are `arg->getNumValues() == 1`. The input file is `InputClass`.

Reply:
This made me revise this area. Before the latest revision, the OP_INPUT check would skip the iteration and we would never hit the ::InputClass case. Now the skip is done by the InputClass.

Reply:
Since this function is pretty isolated and static, can we write unit tests for this function?